### ISSUE TITLE
learning rate update before optimizer.step()

### ISSUE THREAD

**Issue by AlbertiPot** [jadore801120/attention-is-all-you-need-pytorch#188]
Hi, thank you for your wonderful work and codes.

There is one question.

In the Optim.py, [line17,](https://github.com/jadore801120/attention-is-all-you-need-pytorch/blob/132907dd272e2cc92e3c10e6c4e783a87ff8893d/transformer/Optim.py#L17) the learning rate update is executed before the optimizer step, and this may miss the initial lr rate when optimize.

**Comment by AlbertiPot** [comment]
Found initial lr is set by the lr scheduler update function
